### ISSUE TITLE
`SetFocusedOnEntryCompletedBehavior` ignores the `CursorPosition` of the next `Entry` on iOS (not tested others)

### ISSUE THREAD

**Issue by eli191** [CommunityToolkit/Maui#832]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When `CursorPosition` is set to a value greater than 0 on the next `Entry` to be focused, the cursor will fall on the 0 position, ignoring the `CursorPosition` value.
If the `Entry` is manually focused by tapping on it, the `CursorPosition` is working.

### Expected Behavior

The next `Entry` should be focused with the cursor placed on the `CursorPosition` value.

### Steps To Reproduce

You can use the samples [here](https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Behaviors/SetFocusOnEntryCompletedBehaviorPage.xaml) and add a value for `CursorPosition`.

### Link to public reproduction project repository

https://github.com/eli191/MauiCommunityToolkit.CursorPosition

### Environment

```markdown
- .NET MAUI CommunityToolkit: 3.1.0
- OS: iOS 16.0.1418
- .NET MAUI: 7.0.49
```

**Comment by vhugogarcia** [comment]
@eli191 the issue with the cursor position set to 0 has been fixed in .NET MAUI 8, now it defaults to the end of the text length  of the entry. 

However, you pointed out that the CursorPosition property is being ignored when moving to the next field. 
After spending a couple of hours trying to see if we can workaround the issue from the MCT behavior itself, we found out it won't be possible due to the root cause of the issue is on .NET MAUI. If we set the **CursorPosition**  property in the Entry XAML control, it is not being counted to set the position, and it always fall at the end of the text in the entry control. See this [file](https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Platform/iOS/Extensions/TextExtensions.cs) for further references.

Good news! I found a workaround that will solve the issue without a custom handler. See the code sample below:
```xml
<Entry
	x:Name="Entry1"
	Margin="0,0,0,10"
	Placeholder="Entry 1 (Tap `Next` or `Enter` or `Return` when finished)"
	ReturnType="Next"
	Completed="Entry1_Completed"/>
<Entry
	x:Name="Entry2"
	Text="Entry 2"
	Placeholder="Entry 2 (Tap `Next` or `Enter` or `Return` on the keyboard when finished)"
	ReturnType="Next" />
```

```csharp
	void Entry1_Completed(System.Object sender, System.EventArgs e)
	{
		Entry2.CursorPosition = 5;
	}
```

I know it may not be ideal, but this is an alternative to a custom handler. 

@brminnick What do you recommend on this particular scenario? Can we move this ticket to the .NET MAUI repository?, since it this issue does not belong to the `SetFocusedOnEntryCompletedBehavior` itself. 

Please find here below a reproduction of the issue sample project, I updated to .NET MAUI 8 for easy to track an updated project.
[MauiCommunityToolkit.CursorPosition-maui-8.zip](https://github.com/CommunityToolkit/Maui/files/13492056/MauiCommunityToolkit.CursorPosition-maui-8.zip)

**Comment by brminnick** [comment]
> Can we move this ticket to the .NET MAUI repository?

Agreed. @jfversluis do you have permissions to move this Issue to `dotnet/maui`?

**Comment by jfversluis** [comment]
Issues cannot be moved between different GitHub orgs.

@vhugogarcia can I maybe ask you to open this as a new issue on the .NET MAUI repo? You have all the details so you can probably describe it the best. Maybe you even have an idea around a fix since you already pinpointed where it goes wrong? Or else let's work together to get a bug report in there. I think all the talk about the Behavior in this issue might be confusing and not helping the issue get the attention it deserves.

**Comment by vhugogarcia** [comment]
Sounds like a plan @jfversluis let's work on that. I will create the issue there with all the details and attach the sample. Then see if we can fix it together🫡😎

Thanks

**Comment by vhugogarcia** [comment]
The issue has been created on the .NET MAUI repository with fresh information, video demo and repository. https://github.com/dotnet/maui/issues/19109

I will follow-up on the .NET MAUI repository to see if I can contribute there to fix the issue. I'm closing this ticket.

Thank you all!
